### PR TITLE
[SYCL][ESIMD] Add ESIMD tests on constants specialization.

### PR DESCRIPTION
**Pull request by davoroni**

### CODE REVIEW

**Comment by davoroni** [comment]
I made “private/davoroni-esimd-spec-const” branch, merged changes into it, and ran CI with it.

These are results of PI_OpenCL (+2 new esimd tests)

[2020-10-15T13:57:12.310Z] PASS: SYCL :: ESIMD/histogram_256_slm_spec.cpp (51 of 108)
...
[2020-10-15T13:57:23.534Z] PASS: SYCL :: ESIMD/mandelbrot/mandelbrot_spec.cpp (57 of 108)
...
[2020-10-15T13:58:31.418Z]   Unsupported      :  8
[2020-10-15T13:58:31.418Z]   Passed           : 97
[2020-10-15T13:58:31.418Z]   Expectedly Failed:  3

Should I create new PR from “private/davoroni-esimd-spec-const” or we can continue with this PR?

File: SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+template <typename ACC>
+ESIMD_INLINE void mandelbrot(ACC out_image, int ix, int iy, int crunch,
+                             float xOff, float yOff, float scale, float thrs) {
+  ix *= 8;
+  iy *= 2;
+
+  simd<int, 16> m = 0;
+
+  // SIMT_BEGIN(16, lane) {
```
Comment (kbobrovs) [inline SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp]:
```suggestion
```

File: SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      y = x * y * 2.0f + yPos;
+      x = xx - yy + xPos;
+      yy = y * y;
+      xx = x * x;
+      mtemp += 1;
+    } while ((mtemp < crunch) & (xx + yy < thrs));
+
+    m.select<1, 0>(lane) = mtemp;
+
+  } // SIMT_END
```
Comment (kbobrovs) [inline SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp]:
```suggestion
  }
```

**Review by kbobrovs — CHANGES_REQUESTED** [review]

File: SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp (hunk 1 above)
Comment (davoroni) [reply SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp]: done

File: SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp (hunk 2 above)
Comment (davoroni) [reply SYCL/ESIMD/mandelbrot/mandelbrot_spec.cpp]: done

**Review by kbobrovs — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**